replication.go

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"bytes"
+
+	"go.uber.org/zap"
+)
+
+type Request struct {
+	FinalizationCertificateRequest *FinalizationCertificateRequest
+	LatestRoundRequest             *LatestRoundRequest
+}
+
+type Response struct {
+	FinalizationCertificateResponse *FinalizationCertificateResponse
+	LatestRoundResponse             *LatestRoundResponse
+}
+
+// request a finalization certificate for the given sequence number
+type FinalizationCertificateRequest struct {
+	Seq uint64
+}
+
+type FinalizationCertificateResponse struct {
+	FCert FinalizationCertificate
+	Block Block
+}
+
+type LatestRoundRequest struct {
+}
+
+// if block has not been proposed yet, we should return the last round's block, notarization, and fCert
+type LatestRoundResponse struct {
+	Block        Block
+	Notarization *Notarization
+	FCert        *FinalizationCertificate
+}
+
+func (e *Epoch) HandleRequest(req Request) *Response {
```

Comment:
> // TODO: should I update requests to be async? and have the epoch respond with e.Comm.Send(msg, node)

Yeah for now I think it makes sense for the comm to handle this request, and later on we could delegate this to a separate construct.